bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -69,18 +69,20 @@
 public class HomekitImpl implements Homekit, NetworkAddressChangeListener {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
 
+    private final StorageService storageService;
     private final NetworkAddressService networkAddressService;
     private final ConfigurationAdmin configAdmin;
-    private final Storage<String> storage;
+    private final ItemRegistry itemRegistry;
+    private final MetadataRegistry metadataRegistry;
 
-    private HomekitAuthInfoImpl authInfo;
+    private final List<HomekitAuthInfoImpl> authInfos = new ArrayList<>();
```

Comment:
probably in 99,9% of the cases these 4 new ArrayLists will have size of 1.
so, im wondering whether we should initialize them later in constructor with actual size, e.g.
... ArrayList<>(settings.instances)

not sure how much memory we save.. probably not much.
@lolodomo what do you think?

Reply:
I doubt it's worth it. I've been exploring core and the rule engine a lot lately, and there are sooooo many objects allocated, this is barely any noise.

Reply:
> I doubt it's worth it. I've been exploring core and the rule engine a lot lately, and there are sooooo many objects allocated, this is barely any noise.

you are right, we should not try to optimize where it is not required and has very little impact on overall memory usage